zschema/compounds.py

```diff
@@ -427,19 +433,22 @@ def print_indent_string(self):
         for name, field in sorted(self.definition.iteritems()):
             field.print_indent_string(name, 0)
 
-    def validate(self, value, policy=_NO_ARG):
+    def validate(self, value, policy=_NO_ARG, path=_NO_ARG):
         if policy is None:
             policy = _NO_ARG
+        if not path:
+            path = []
         calculated_policy = self._calculate_policy("root", policy, self.validation_policy)
         # ^ note: record explicitly does not take a parent_policy
         if type(value) != dict:
-            raise DataValidationException("record is not a dict:\n{}".format(value))
+            raise DataValidationException("record is not a dict:\n{}".format(value), path=path)
         for subkey, subvalue in sorted(value.iteritems()):
+            print '!', subkey, subvalue
```

Comment:
Is this debugging code that snuck into a commit?

Reply:
yikes